testplan/common/entity/base.py

```diff
@@ -253,18 +236,17 @@ def stop(self, is_reversed=False):
                 )
                 resource.logger.error(msg)
                 self.stop_exceptions[resource] = msg
+                # Resource status should be STOPPED even it failed to stop
+                resource.wait(resource.STATUS.STOPPED)
```

Comment:
I saw this has been changed to wait(), which I do not think is right. Because stop() has exceptioned so no one is changing the status to STOPPED, so wait here might just hang？

Also, if resource failed to stop, what status should it have? Maybe we don't bother to change the status here in the first place. Because it's meant to be kind of undefined, it's not really STOPPED right？

Reply:
Kris had a comment at the same place, but he is concerning the private access of resource. I left a comment there for this issue.